pkg/querier/queryrange/log_result_cache.go

```diff
@@ -105,7 +108,17 @@ func (l *logResultCache) Do(ctx context.Context, req queryrangebase.Request) (qu
 	// The first subquery might not be aligned.
 	alignedStart := time.Unix(0, lokiReq.GetStartTs().UnixNano()-(lokiReq.GetStartTs().UnixNano()%interval.Nanoseconds()))
 	// generate the cache key based on query, tenant and start time.
-	cacheKey := fmt.Sprintf("log:%s:%s:%d:%d", tenant.JoinTenantIDs(tenantIDs), req.GetQuery(), interval.Nanoseconds(), alignedStart.UnixNano()/(interval.Nanoseconds()))
+
+	transformedTenantIDs := tenantIDs
+	if l.transformer != nil {
+		transformedTenantIDs = make([]string, 0, len(tenantIDs))
+
+		for _, tenantID := range tenantIDs {
+			transformedTenantIDs = append(transformedTenantIDs, l.transformer(ctx, tenantID))
+		}
+	}
+
+	cacheKey := fmt.Sprintf("log:%s:%s:%d:%d", tenant.JoinTenantIDs(transformedTenantIDs), req.GetQuery(), interval.Nanoseconds(), alignedStart.UnixNano()/(interval.Nanoseconds()))
```

Comment:
I still don't fully understand the name to transform IDs either here or in the results cache. where can I find a reproduction of the bug this is causing?

Reply:
sent you the code reproducing the issue in our internal slack thread